assets/ts/stop/components/DeparturesAndMap.tsx

```diff
@@ -157,11 +157,15 @@ const DeparturesAndMap = ({
   // filter by chosen route and direction
   const filteredDepartures = activeRow
     ? updatedDepartureInfos.filter(departure => {
-        const { route, trip } = departure;
+        const { route, schedule, trip } = departure;
         return (
           route.id === activeRow.route.id &&
           trip.direction_id === activeRow.directionId &&
-          trip.headsign === activeRow.headsign
+          // We have to match the headsign from the schedule or the trip.
+          // If the schedule doesn't have a headsign, we use the trip's headsign.
+          (schedule?.stop_headsign === activeRow.headsign ||
+            (schedule?.stop_headsign === null &&
+              trip.headsign === activeRow.headsign))
```

Comment:
```suggestion
          (schedule?.stop_headsign === activeRow.headsign ||
            trip.headsign === activeRow.headsign)
```
FYI if `schedule` is undefined (this may happen for predictions with `schedule_relationship=added`) then `schedule?.stop_headsign === null` will be false as `schedule?.stop_headsign` will resolve to `undefined`. Anyway, am not convinced you need to check that in this line anyway.